ui/src/app/base/components/TitledSection/TitledSection.tsx

```diff
@@ -1,31 +1,67 @@
+import React from "react";
 import type { ReactNode } from "react";
 
 import { Strip } from "@canonical/react-components";
-import type { PropsWithSpread, StripProps } from "@canonical/react-components";
+import type {
+  PropsWithSpread,
+  StripProps,
+  Headings,
+} from "@canonical/react-components";
+
+import VisuallyHidden from "../VisuallyHidden";
 
 import { useId } from "app/base/hooks/base";
 
 type Props = PropsWithSpread<
   {
     buttons?: ReactNode;
     children?: ReactNode;
+    headingElement?: Headings;
+    headingClassName?: string;
+    headingVisuallyHidden?: boolean;
     title: ReactNode;
   },
   StripProps
 >;
 
+type HeadingProps = {
+  element: Headings;
+  className?: string;
+  id: string;
+  children: ReactNode;
+};
+
+const Heading = ({ element, id, className, children }: HeadingProps) =>
+  React.createElement(
+    element,
+    {
+      id,
+      className,
+    },
+    children
+  );
+
 const TitledSection = ({
   buttons,
   children,
   title,
+  headingElement = "h2",
+  headingClassName = "p-heading--4",
+  headingVisuallyHidden, // hide the title visually (visibly hidden but still accessible)
```

Comment:
I'm curious what the benefit of this is over setting the `aria-label` on the section?

Reply:
It's still a heading and provides semantics that are more suitable in this context as this is a section (there's a clear difference between a label and headings, e.g. with screen readers you can quickly go through the entire document via headings).

Also consistency - as a dev user of this component you'd expect that you'll get a section with a title (heading), not just a label.